### PR TITLE
Renovation: refactor endUpdate method of base components

### PR DESCRIPTION
**Pull request by LazyLahtak**
Extract logic from the `endUpdate` method into separate methods.

### CODE REVIEW

File: js/core/component.js [hunk 1]
```diff
@@ -192,6 +192,46 @@ var Component = Class.inherit({
         this._disposed = true;
     },
 
+    _lockUpdate() {
+        this._updateLockCount++;
+    },
+
+    _removeLockUpdate() {
```
Comment (churkin) [inline js/core/component.js]: `unlock`?

File: js/core/component.js [hunk 2]
```diff
@@ -192,6 +192,46 @@ var Component = Class.inherit({
...
+    },
+
+    _commitUpdate() {
+        this.postponedOperations.callPostponedOperations();
+        if(this._isInitializingRequired()) {
+            this._commitInit();
+        }
+    },
+
+    _commitInit() {
```
Comment (churkin) [inline js/core/component.js]: bad method's name

File: js/core/dom_component.js [hunk 3]
```diff
@@ -191,21 +191,21 @@ const DOMComponent = Component.inherit({
     },
 
     _attachDimensionChangeHandlers() {
-        const $el = this.$element();
+        const $element = this.$element();
```
Comment (churkin) [inline js/core/dom_component.js]: I use this short name :)

**Review by churkin — CHANGES_REQUESTED** [review]

File: js/core/dom_component.js (hunk 3 above)
Comment (LazyLahtak) [reply js/core/dom_component.js]: In this component other methods use name `$element`. Should we change all `$element` to `$el`?

File: js/core/component.js (hunk 2 above)
Comment (LazyLahtak) [reply js/core/component.js]: What about `_initializeComponent`?

File: js/core/dom_component.js (hunk 3 above)
Comment (LazyLahtak) [reply js/core/dom_component.js]: I think I'll rollback these changes and will refactor this component in other PR

**Comment by LazyLahtak** [comment]
Ready to review

**Review by churkin — DISMISSED** [review]

**Review by MaximKudriavtsev — DISMISSED** [review]

File: js/core/dom_component.js [hunk 4]
```diff
@@ -404,19 +404,20 @@ const DOMComponent = Component.inherit({
             .join(' ');
     },
 
-    endUpdate() {
-        const requireRender = !this._initializing && !this._initialized;
+    _updateDOMComponent(requireRender) {
```
Comment (IlyaKhD) [inline js/core/dom_component.js]:
```suggestion
    _updateDOMComponent(renderRequired) {
```

File: js/core/dom_component.js [hunk 5]
```diff
@@ -404,19 +404,20 @@ const DOMComponent = Component.inherit({
...
-        if(!this._updateLockCount) {
-            if(requireRender) {
-                this._renderComponent();
-            } else if(this._requireRefresh) {
-                this._requireRefresh = false;
-                this._refresh();
-            }
-        }
+        this.callBase();
+        this._isUpdateAllowed() && this._updateDOMComponent(requireRender);
```
Comment (IlyaKhD) [inline js/core/dom_component.js]:
```suggestion
        this._isUpdateAllowed() && this._updateDOMComponent(this._isInitializingRequired());
```

Comment (LazyLahtak) [reply js/core/dom_component.js]: We can't use `this._isInitializingRequired()` here.
After `this.callBase();` we get other values of `this._initializing` and `this._initialized` and other value of `this._isInitializingRequired()`.

**Review by nightskylark — APPROVED** [review]

**Pull request merged**